src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -250,11 +248,30 @@ trait PodsValidation extends GeneralPurposeCombinators {
     pod is endpointNamesUnique and endpointContainerPortsUnique and endpointHostPortsUnique
     pod should complyWithPodUpgradeStrategyRules
     pod should haveUnreachableDisabledForResidentPods
-    pod should haveValidAcceptedResourceRoles
+    pod should haveValidAcceptedResourceRoles(roleSettings.validRoles)
+    pod is validWithRoleEnforcement(roleSettings)
+  }
+
+  private def validWithRoleEnforcement(roleSettings: RoleSettings): Validator[Pod] = validator[Pod] { pod =>
+    if (roleSettings.enforceRole) {
+      pod.role must notEmpty
```

Comment:
This reads a little odd. I don't think that `enforceRole` is about requiring the role to be set but that the role must be the one from the top-level group. @timcharper could you clarify?

Reply:
Yeah, this was because the validation was done on the PodRaml, and before normalization was done. I've now changed it to validate the role only on the PodDefinition and after normalization.
At some point it might be a good idea to do validation/normalization with the same pattern on apps and pods...